### DISCUSSION TITLE
Save "pbi to files" into relative path

### DISCUSSION THREAD

**Discussion by hlynurarni** [didierterrien/Power-BI-Sidetools#23] [Ideas]
Just thought of a feature/request that I think could benefit many.

- Be able to save pbi to files to the same directory as the pbi file is located. Eliminates doing the manual step of selecting the folder or saving into wrong place.
- EDIT: In addition to the relative directory it would have to be optional to change the name of the folder that is created, currently nothing happens because there is a .pbix file with the same name in that folder.

![sidetools_feature](https://user-images.githubusercontent.com/10733420/125275191-02726c80-e30f-11eb-8213-15c17fdea104.png)

**Comment by didierterrien** [comment]
That's an interresting idea. I will think about possible side effects and will add to TODO list.
Thank you for your contribution

**Reply by hlynurarni** [reply]
Thank you for the quick reply and great work on the tool. 

To give better context, we aim to use the tool for version and hopefully source control.
We would push the files coming from "pbi to files" into git and handle the .pbix as large files storage (lfs).

**Comment by didierterrien** [comment]
I added a setting to handle this case. You can try it in the V1.2.5 test version :
[https://github.com/didierterrien/Power-BI-Sidetools/tree/master/Power%20BI%20Sidetools%20test%20versions](https://github.com/didierterrien/Power-BI-Sidetools/tree/master/Power%20BI%20Sidetools%20test%20versions)

**Comment by didierterrien** [comment]
According to #24 issue, you tested the new feature.
As the modification involved critical part of the code, I will continue to test it before publishing. Let me know if it's robust from your side.

**Reply by hlynurarni** [reply]
This works really well as far as I have tested it. Will update if I encounter any issues. Great work ! 

Having this option along with Git will changed a lot on how we work with .pbix files. There are probably some ways to make this process more seamless in the future.

**Comment by hlynurarni** [comment]
One thing that I would like to point out. 

Before the update the write access to the folder Sidetools creates was locked by the application. Meaning I could not delete the folder without getting an error if I had Sidetools running. Now I can have the connection established and delete the folder without issues. 

For me this doesn't change anything and I like that Sidetools just creates a new folder if it doesn't exist, like nothing happened. 

Just wanted to let you know if you had overlooked that change in the code and it could create conflicts later on.

BR Hlynur

**Comment by didierterrien** [comment]
Hello Hlynur,
I don't have the same behaviour. From my side, the folder is locked. Anyway, I suppose it is not a big deal if the user does it intentionally. Thank you for mentionning it ! I will keep it in mind for future developments.

**Comment by didierterrien** [comment]
Power BI Sidetools has been updated to version 1.2.5
